File: frontend/lib/models/expense.dart
class Expense{
  int? id = 0;
  String name = "";
  String date = "";
  double amount = 0;
  String description = "";
  int periodicityNumber = 0;

  Expense(
    this.name,
    this.date,
    this.amount,
    this.description,
    this.periodicityNumber
  );

  Expense.fromJson(Map<String, dynamic> json){
    id = json['id'];
    date = json['date'];
    name = json['name'];
    amount = json['amount'].toDouble();
    description = json['description'];
    periodicityNumber = json['periodicityNumber'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};

    data['id'] = id;
    data['date'] = date;
    data['name'] = name;
    data['amount'] = amount;
    data['description'] = description;
    data['periodicityNumber'] = periodicityNumber;

    return data;
  }
}

File: backend/src/Entity/Expense.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UnderPantsApp.Entity
{
    public class Expense
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public int PeriodicityNumber { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public double Amount { get; set; }

        [Required]
        [MaxLength(255)]
        public string Description { get; set; } = string.Empty;

        [Required]
        [MaxLength(255)]
        public string Name { get; set; } = string.Empty;

    }
}